### PR TITLE
add macos_arm64 exe

### PR DESCRIPTION
**Pull request by e-perl-NOAA**
Add SS3 executable for the arm64 version of macs.

### CODE REVIEW

File: R/runSS.R [hunk 1]
```diff
@@ -65,7 +65,7 @@ run_ss_model <- function(dir,
   if (os == "unix") {
     system(
       paste0(
-        "cd ", dir, ";", paste0(bin, " "),
+        "sudo chmod a+x ", bin, "; ", "cd ", dir, "; ", paste0(bin, " "),
```
Comment (k-doering-NOAA) [inline R/runSS.R:68]: @e-perl-NOAA was this needed to guarantee that the permissions were set correctly?

Or could we change the permissions on the file within the repo only?

**Review by k-doering-NOAA — CHANGES_REQUESTED** [review]
Looks good! I had one comment that I'm not sure is relevant.

I also would like to squash the 5 commits before rebasing this commit into main, so I plan to do that once the comment is addressed.

File: R/runSS.R (hunk 1 above)
Comment (e-perl-NOAA) [reply R/runSS.R:68]: I needed it to get the github action that runs the tests on the macos-14 runner to pass. However, I think to absolutely make sure the permissions are correct it wouldn't necessarily be a bad thing to have it in there, especially for macs. I feel like I've had to tell people often enough with macs to make sure the change the permission for the exe.

Comment (k-doering-NOAA) [reply R/runSS.R:68]: yes, that's very true, I remember telling people that a lot as well. I think it's good to stay as-is then, thanks!

**Review by k-doering-NOAA — APPROVED** [review]

**Pull request merged**